ios/FluentUI/Tooltip/Tooltip.swift

```diff
@@ -50,55 +50,59 @@ open class Tooltip: NSObject, TokenizedControlInternal {
+                                                           tokenSet: tokenSet)
+        if let tooltipViewController = tooltipViewController,
+           let tooltipView = tooltipViewController.view,
+           let rootViewController = window.rootViewController {
+            let rootView = rootViewController.view
+            rootViewController.addChild(tooltipViewController)
+            tooltipView.accessibilityViewIsModal = true
+            self.onTap = onTap
+            self.dismissMode = UIAccessibility.isVoiceOverRunning ? .tapOnTooltip : dismissMode
+            let gestureView = TouchForwardingView(frame: window.bounds)
+            self.gestureView = gestureView
+            switch self.dismissMode {
+            case .tapAnywhere:
+                rootView?.addSubview(tooltipView)
+                rootView?.addSubview(gestureView)
+                gestureView.onTouches = { _ in
                     self.handleTapGesture()
                 }
+            case .tapOnTooltip, .tapOnTooltipOrAnchor:
+                rootView?.addSubview(gestureView)
+                rootView?.addSubview(tooltipView)
+                gestureView.forwardsTouches = false
+                tooltipView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(handleTapGesture)))
+                if self.dismissMode == .tapOnTooltipOrAnchor {
+                    gestureView.passthroughView = anchorView
+                    gestureView.onPassthroughViewTouches = { _ in
+                        self.handleTapGesture()
+                    }
+                }
             }
-        }
 
-        // Layout tooltip
-        tooltipView.frame = positionController.tooltipRect
+            tooltipViewController.didMove(toParent: rootViewController)
 
-        // Animate tooltip
-        tooltipView.alpha = 0.0
-        UIView.animate(withDuration: Constants.animationDuration, delay: 0.0, options: [.curveEaseOut], animations: {
-            tooltipView.alpha = 1.0
-        }, completion: { _ in
-            UIAccessibility.post(notification: .screenChanged, argument: tooltipView)
-        })
+            // Layout tooltip
+            tooltipView.frame = positionController.tooltipRect
+
+            // Animate tooltip
+            tooltipView.alpha = 0.0
+            UIView.animate(withDuration: Constants.animationDuration, delay: 0.0, options: [.curveEaseOut], animations: {
+                tooltipView.alpha = 1.0
+            }, completion: { _ in
+                UIAccessibility.post(notification: .screenChanged, argument: tooltipView)
+            })
 
-        isShowing = true
+            isShowing = true
 
-        UIDevice.current.beginGeneratingDeviceOrientationNotifications()
-        NotificationCenter.default.addObserver(self, selector: #selector(orientationChanged), name: UIDevice.orientationDidChangeNotification, object: nil)
+            UIDevice.current.beginGeneratingDeviceOrientationNotifications()
```

Comment:
none of this code should be needed any more with your awsome new viewwillresize?

Reply:
oh I thought you didn't want to change existing behavior. I'll update it to reposition instead of hide

Reply:
you cdan still make it dismissed but you don't need this observer anymore?

Reply:
before they needed this hack because it was just a uiview. now that you have converted to viewcontroller there is much better way to tell the orientation has changed

Reply:
gotcha. that's a bit tricky because viewWillTransitionTo gets called for screen rotation _and_ window resizing. there's not really a clean way to determine whether the method was called due to rotating or multi-task resizing. when viewWillTransitionTo gets called, UIDevice.current.orientation has already updated to the new orientation. otherwise i would compare the orientation outside the coordinator completion with the orientation within the coordinator completion. we could store a variable that keeps track of the orientation to compare, but i think the existing implementation with observers is better

Reply:
the UIDevice.current is as wrong practice as uiapplicationshared. iPad now can share their screen with monitor so automatically this usuage is incorrect. the traitcollectiondidchange can be another place to figure out the rotation happened to check before and after state?

Reply:
yup! traitcollectiondidchange works (though that still worked when it was just uiview too)

Reply:
if we just check the currently view bounds height to the size that will be transition height, if the height changes, does that mean it is rotated? in window size changes the height stays the same and usual the width changes.

Reply:
seems hacky to me, there are edge cases to that. for example, changing the app from slide over to split view and vice versa changes the window height too

Reply:
thanks! i see. then i think to remove all the hackiness together, lets just reposition the tooltip all the time rather than guess the rotation.

Reply:
sounds good! :)